.travis.yml

```diff
@@ -1,44 +1,84 @@
-      NUMPY_VERSION="1.15.1" SCIPY_VERSION="1.1.0" PANDAS_VERSION="0.23.4"
-      SIX_VERSION="1.10.0" COVERAGE="true"
-    - DISTRIB="conda" PYTHON_VERSION="3.5" INSTALL_MKL="true"
-      NUMPY_VERSION="1.15.1" SCIPY_VERSION="1.1.0" PANDAS_VERSION="0.23.4"
-      SIX_VERSION="1.10.0" COVERAGE="true"
-    # This environment tests minimal dependency versions
-    - DISTRIB="conda_min" PYTHON_VERSION="2.7" INSTALL_MKL="false"
-      SIX_VERSION="1.10.0" NUMPY_VERSION="1.8.2" SCIPY_VERSION="0.14.0" COVERAGE="true"
-    - DISTRIB="conda_min" PYTHON_VERSION="3.4" INSTALL_MKL="false"
-      SIX_VERSION="1.10.0" NUMPY_VERSION="1.8.2" SCIPY_VERSION="0.14.0" COVERAGE="true"
-    # basic Ubuntu build environment
-    - DISTRIB="ubuntu" PYTHON_VERSION="2.7" INSTALL_ATLAS="true"
-      COVERAGE="true"
-    # This environment tests for mpi
-    - DISTRIB="mpi" PYTHON_VERSION="3.5" INSTALL_MKL="false"
-      NUMPY_VERSION="1.15.1" SCIPY_VERSION="1.1.0" SIX_VERSION="1.10.0"
-      MPI_VERSION="2.0.0" COVERAGE="true" MPI="true"
-
-install: source continuous_integration/install.sh
-script: bash continuous_integration/test_script.sh
+     update: true
+
+
+matrix:
+  include:
+    - name: "conda 2.7"
+      python: 2.7
+      env: DISTRIB="conda"
+
+    - name: "pip 2.7"
+      python: 2.7
+      env: DISTRIB="pip"
+
+    - name: "conda 3.5"
+      python: 3.5
+      env: DISTRIB="conda"
+
+    - name: "conda 3.6"
+      python: 3.6
+      env: DISTRIB="conda"
+
+    - name: "pip 3.6"
+      python: 3.6
+      env: DISTRIB="pip"
+
+    - name: "pip 3.6 requirements-extras"
+      python: 3.6
+      env: DISTRIB="pip"
+      before_install: sudo apt install -y libopenmpi-dev openmpi-bin
+      before_script: pip install -r requirements-extras.txt
+      script: mpiexec -n 1 nosetests --with-coverage --cover-package=elephant
+
+    - name: "conda 3.7"
+      python: 3.7
+      env: DISTRIB="conda"
+
+  exclude:
+    - name: "pip 3.6 requirements min version"
+      # excluded due to unmet dependencies in neo, quantities, and scipy
+      python: 3.6
+      env: DISTRIB="pip"
+      before_install:
+        - sudo apt install libblas-dev liblapack-dev libatlas-base-dev gfortran
+        - sed -i 's/>=/==/g' requirements.txt
+
+
+install:
+  - if [[ "${DISTRIB}" == "conda" ]];
+    then
+      export MINICONDA_PATH=${HOME}/miniconda;
+      export PATH=${MINICONDA_PATH}/bin:${PATH};
+      wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
+      bash miniconda.sh -b -p ${MINICONDA_PATH};
+      conda config --set always_yes yes;
+      conda update conda;
+      conda create --name elephant_env python=${TRAVIS_PYTHON_VERSION};
+      source activate elephant_env;
+      conda install mkl;
+      conda config --append channels conda-forge;
+      sed '/^neo/d' requirements.txt > requirements-conda.txt;
+      conda install --file requirements-conda.txt;
+      conda install -c conda-forge python-neo;
```

Comment:
This would, to my understanding, install always the newest Neo version from forge for conda environments, however, requirements will typically indicate a fixed major version of neo, e.g.,
`neo>=0.7.1,<0.8.0`
I think it may be better to either install only the neo package via pip similar to the `sed` trick above, or to specify the neo version on the conda install line.

Reply:
You are right - it'd be better to leave neo in `requirements-conda.txt`. However, python-neo conda version is outdated and not supported very well. If you run `conda install -c conda-forge neo>=0.7.1` it'll throw you an error indicating that it could not find a package with such a version because the latest one is 0.7.0.

Reply:
Yes, in fact I am not even sure if the forge neo package is actually done by the project, or by someone outside. So better to stick to pip probably...

Reply:
I found it here https://github.com/NeuralEnsemble/python-neo/issues/233.
OK, I'll replace this line to `pip install neo>=...`